### PR TITLE
Admin signup

### PR DESCRIPTION
**Pull request by cem-gurel47**

### CODE REVIEW

File: back-end/index.js [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
 const firebase = require("firebase/app");
 const app = express();
 const port = process.env.PORT || 5000;
 const indexRouter = require("./routes/routes");
 const authRouter = require("./routes/authRouter");
 const eventRouter = require("./routes/eventRoutes");
-
+const verifyToken = require("./middlewares/verifyToken");
 const cors = require("cors");
+const serviceAccount = require("./adminCredentials.json");
```
Comment (Michael-Kagnew) [inline back-end/index.js:12]: I believe this is already in the environment variables, or are we also using files with sensitive info?

**Review by hamzahabouseta — APPROVED** [review]

**Review by tyrellshawn — APPROVED** [review]
Nice work Cem!

File: back-end/middlewares/verifyToken.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const admin = require("firebase-admin");
+
+const verifyToken = async (req, res, next) => {
+  try {
+    // don't need to check for token when user is trying to register
+    if (!req.url?.startsWith("/auth")) {
+      const token = req.headers.authorization;
+      const decodedToken = await admin.auth().verifyIdToken(token);
+      req.user = decodedToken; //now we can use req.user to get the user object
```
Comment (Michael-Kagnew) [inline back-end/middlewares/verifyToken.js:9]: Maybe make this part of the request body, instead of a new object?

**Review by Michael-Kagnew — COMMENTED** [review]
Few comments left

File: back-end/routes/authRouter.js [hunk 3]
```diff
@@ -41,4 +42,17 @@ authRouter.post("/register", async (req, res) => {
     );
 });
 
+authRouter.post("/claim-user-admin", async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Michael-Kagnew) [reply back-end/routes/authRouter.js:48]: Route moved to a new route group, resolved

**Review by Michael-Kagnew — APPROVED** [review]
Had only one minor suggestion left, but LGTM

File: back-end/middlewares/verifyToken.js (hunk 2 above)
Comment (Michael-Kagnew) [reply back-end/middlewares/verifyToken.js:9]: This is not an issue, once logged in Firebase automatically puts user in the headers of all the requests. Leave as is

**Pull request merged**